lib/msf/core/exploit/browser_autopwnv2.rb

```diff
@@ -0,0 +1,834 @@
+      start_exploits
+
+      print_status("Starting listeners...")
+      start_payload_listeners
+
+      t2 = Time.now
+      print_status("Time spent: #{(t2-t1).inspect}")
+    end
+
+
+    # Prints all the exploits that BAP will consider using. But this isn't the actual list of
+    # exploits that BAP will use for each target.
+    #
+    # @return [void]
+    def show_ready_exploits
+      columns = ['Order', 'Rank', 'Name', 'Path', 'Payload']
+
+      # If not verbose, you're not in dev mode.
+      # As an user, you shouldn't be using any of these paths anyway.
+      columns.delete('Path') if !datastore['VERBOSE']
+
+      table = Rex::Ui::Text::Table.new(
+        'Header'  => 'Exploits',
+        'Indent'  => 1,
+        'Columns' => columns
+      )
+
+      # Without the order, sometimes the Rex table messes up even though in the array
+      # the order looks right. So don't get rid of this.
+      order = 1
+
+      bap_exploits.each do |m|
+        row = []
+        row << order
+        row << parse_rank(m.rank)
+        row << m.shortname
+        row << m.datastore['URIPATH'] if datastore['VERBOSE']
+        row << "#{m.datastore['PAYLOAD']} on #{m.datastore['LPORT']}"
+        table << row
+        order += 1
+      end
+
+      print_line
+      print_status("The following is a list of exploits that BrowserAutoPwn will consider using.")
+      print_status("Exploits with the highest ranking and newest will be tried first.")
+      print_line
+      print_line table.to_s
+    end
+
+
+    # Prints information such as what exploits will be used, and the BAP URL.
+    #
+    # @return [void]
+    def start_service
+      super
+      show_ready_exploits
+      proto = (datastore['SSL'] ? "https" : "http")
+      srvhost = (datastore['SRVHOST'] == '0.0.0.0') ? Rex::Socket.source_address : datastore['SRVHOST']
+      srvport = datastore['SRVPORT']
+      service_uri = "#{proto}://#{srvhost}:#{srvport}#{get_resource}"
+      print_status("Please use the following URL for the browser attack:")
+      print_status("BrowserAutoPwn URL: #{service_uri}")
+    end
+
+
+    # Prints all user-configurable payloads. It's the same as the "show payloads" command in console.
+    #
+    # @return [void]
+    def show_payloads
+      DEFAULT_PAYLOADS.keys.each do |platform|
+        payload_name = get_selected_payload_name(platform)
+        p = framework.payloads.create(payload_name)
+        next unless p
+        p.datastore['LHOST'] = get_payload_lhost
+        p.datastore['LPORT'] = get_selected_payload_lport(platform)
+        p_opt = Serializer::ReadableText.dump_options(p, '   ')
+        print("\nPayload options (#{payload_name}):\n\n#{p_opt}\n") if (p_opt and p_opt.length > 0)
+      end
+    end
+
+
+    # Prints a message that explains how the user should set a payload. This is the same as the
+    # "set payload" command in console.
+    #
+    # @return [void]
+    def set_payload
```

Comment:
An easy testcase for this is you create an auxiliary module, include BES, and see what happens. @egypt and I tried to make it work with some hack-ish code... it was still pretty broken so we gave up on that idea.